src/battle/star/up_and_away/7952E0.c

```diff
@@ -43,23 +43,28 @@ ApiStatus func_802A15B4_795894(Evt* script, s32 isInitialCall) {
     return ApiStatus_BLOCK;
 }
 
-#ifdef NON_EQUIVALENT
 ApiStatus func_802A1628_795908(Evt* script, s32 isInitialCall) {
-    Actor* targetActor = get_actor(get_actor(script->owner1.actorID)->targetActorID);
-    s32 flags = targetActor->flags;
-
+    Actor* targetActor = get_actor((s32)get_actor(script->owner1.actorID)->targetActorID);
+    
     script->varTable[0] = 0;
 
-    if ((flags & ACTOR_FLAG_TARGET_ONLY) == NULL && (flags & ACTOR_FLAG_NO_DMG_APPLY) == NULL && (flags & ACTOR_FLAG_2000) == NULL && targetActor->actorBlueprint->upAndAwayChance != 0 && targetActor->actorBlueprint->upAndAwayChance >= rand_int(100)) {
-        script->varTable[0] = 1;
+    {
```

Comment:
Is there a reason for the extra scoping?

Reply:
The script->varTable[0] = 0; line needs to come before the flag declaration, but declaring the flags variable afterwards gives a parsing error.  Putting it in another scope fixes the error.  I'm more used to C++ than C so I don't know of any other way to fix this, but if you know how to, then let me know.

Reply:
Yeah, this is a c89 oddity (all variables need to be declared at the beginning of a scope). Are you sure you can't define the var with no assignment and then assign it after the script->varTable[0] assignment?

Reply:
Oh, yeah that worked.  I've no idea why I didn't try that lol.